evalml/pipelines/components/transformers/samplers/base_sampler.py

```diff
@@ -0,0 +1,56 @@
+from evalml.pipelines.components.transformers import Transformer
+from evalml.utils.woodwork_utils import (
+    _convert_woodwork_types_wrapper,
+    infer_feature_types
+)
+
+
+class BaseSampler(Transformer):
```

Comment:
Added this base class to expand all of our future under/oversamples with.

Reply:
I think this is a solid add and we'll reap the rewards of it later.